k9mail/src/main/java/com/fsck/k9/Account.java

```diff
@@ -1201,10 +1201,7 @@ public synchronized boolean setFolderSyncMode(FolderMode syncMode) {
         if (syncMode == FolderMode.NONE && oldSyncMode != FolderMode.NONE) {
             return true;
         }
-        if (syncMode != FolderMode.NONE && oldSyncMode == FolderMode.NONE) {
-            return true;
-        }
-        return false;
+        return syncMode != FolderMode.NONE && oldSyncMode == FolderMode.NONE;
```

Comment:
If you combine the last two `return` statements you might as well only have one. However, I think the old way makes the code easier to read/understand.